ubuntu/start.sh

```diff
@@ -54,11 +54,20 @@ trap 'cleanup; exit 130' INT
 trap 'cleanup; exit 143' TERM
 
 echo Determining matching VSTS agent...
-VSTS_AGENT_URL=$(curl -LsS \
+VSTS_AGENT_RESPONSE=$(curl -LsS \
```

Comment:
I don't think it needs to be this complicated. There was one bug I discovered, which is that if there was any error, the script would immediately exit due to the "set -e" above. That was unintended, because the code at line 62 was supposed to be handling all error cases and wouldn't be called. This is easily fixed by temporarily doing "set +e" before trying to get the URL and then "set -e" again after. The other thing we want to do is hide errors from jq, so that's a simple "2>/dev/null". Lastly, we just want to update the error message to be more informative. All in all, something like this:

```
echo Determining matching VSTS agent...
set +e
VSTS_AGENT_URL=$(curl -LsS \
  -u user:$(cat "$VSTS_TOKEN_FILE") \
  -H 'Accept:application/json;api-version=3.0-preview' \
  "https://$VSTS_ACCOUNT.visualstudio.com/_apis/distributedtask/packages/agent?platform=linux-x64" \
  | jq -r '.value | map([.version.major,.version.minor,.version.patch,.downloadUrl]) | sort | .[length-1] | .[3]' 2>/dev/null)
set -e
if [ -z "$VSTS_AGENT_URL" -o "$VSTS_AGENT_URL" == "null" ]; then
  echo 1>&2 error: could not determine a matching VSTS agent - check that account \'$VSTS_ACCOUNT\' is correct and the token is valid for that account
  exit 1
fi
```

Reply:
Thanks for the input. I wasnt aware of this bash pattern. The only problem I encountered when testing this updated script is a curl error leaks through ( (23) Failed writing body, `jq` closes pipe before curl is done writing ) , when running with an invalid token

Silencing curl errors is obviously an option but there may be some errors that should actually be shown. 

Instead I added `--slurp` and `--raw-input` to `jq` which seems to have the effect of reading in the entire input into memory before filtering.  Seems to work fine for our circumstances. Let me know what you think.